src/native/xamarin-debug-app-helper/debug-app-helper.cc

```diff
@@ -70,8 +70,12 @@ Java_mono_android_DebugRuntime_init (JNIEnv *env, [[maybe_unused]] jclass klass,
 	const char *monosgen_path = get_libmonosgen_path ();
 	void *monosgen = dlopen (monosgen_path, RTLD_LAZY | RTLD_GLOBAL);
 	if (monosgen == nullptr) {
-		log_fatal (LOG_DEFAULT, "Failed to dlopen Mono runtime from %s: %s", monosgen_path, dlerror ());
-		Helpers::abort_application ();
+		char *message = Util::monodroid_strdup_printf (
+			"Failed to dlopen Mono runtime from %s: %s",
```

Comment:
Given that the assert message length is 120 characters, and that this contains a *path* that could be quite long, e.g. my `adb logcat` mentions `/data/app/~~TeyVIg4zlc0W0sLrmB8kMQ==/com.microsoft.office.outlook-OiHk4wyhwFjmw4C999CCbw==/lib/arm64/libhermes.so` (113 characters), we may want to *reverse* these parameters:

```
"Failed to dlopen MonoVM: %s (from `%s`)", dlerror(), monosgen_path
```